### PR TITLE
feat: adds label-based canary releases

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task](https://app.asana.com/0/1100423001970639/1200828549856250/f)
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

Adds label based canary releases for PRs, see https://github.com/hashicorp/nextjs-scripts/pull/204 for some additional details on the setup!

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: .changeset/config.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
   "updateInternalDependencies": "patch",
   "ignore": [],
   "___experimentalUnsafeOptions_WILL_CHANGE_IN_PATCH": {
```
Comment (zchsh) [inline .changeset/config.json:13]: 😯 Ahah looked for docs on `useCalculatedVersionForSnapshots` and I guess this explains why my Google search came up empty! 👍

File: README.md [hunk 2]
```diff
@@ -42,6 +42,14 @@ To make any adjustments to your changeset, just edit the file!
 
 The release process is handled mostly automatically via the changesets GitHub action. When changeset files get merged to `main`, a Pull Request is opened which will, upon merge, release all pending changesets and remove the changeset files. We should not need to publish manually with this flow. See the `changesets/action`(https://github.com/changesets/action) repo for more information.
 
+### Canary Releases
+
+If you want to test your changes before merging, you can add a `release:canary` label to your pull request. If any changeset files are found, a release will be created and tagged with `canary`. You can then install the canary version elsewhere:
+
+```
+npm instal @hashicorp/react-package@canary
```
Comment (zchsh) [inline README.md:50]:
```suggestion
npm install @hashicorp/react-package@canary
```

**Review by zchsh — COMMENTED** [review]
Tested this out with https://github.com/hashicorp/react-components/pull/321 and the release process seems to have worked pretty well - though I am seeing a ton of unexpected canary releases. Not sure if that's related to not basing the branch on `main` or some other quirk, but wanted to mention just in case.

File: README.md (hunk 2 above)
Comment (BRKalow) [reply README.md:50]: 🤦 thanks!

**Comment by BRKalow** [comment]
@zchsh Yeah so right now this looks at all changeset files that currently exist in the repo, so not necessarily scoped to only the ones from a specific PR. That could be a nice future enhancement, to try and only consider changeset files from the PR.

File: .changeset/config.json (hunk 1 above)
Comment (BRKalow) [reply .changeset/config.json:13]: 😄 Info here: https://github.com/hashicorp/nextjs-scripts/pull/204#discussion_r697616836

**Review by zchsh — APPROVED** [review]
Sweet, that makes sense 👍 💯

File: .changeset/config.json (hunk 1 above)
Comment (zchsh) [reply .changeset/config.json:13]: oh nice, missed that earlier 🙇 ty!

**Pull request merged**